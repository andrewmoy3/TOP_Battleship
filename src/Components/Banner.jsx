import React from "react";
import '../css/Banner.css'

export default function Banner(){
    return (
        <div className="banner" id="banner">
            <div id="signature" data-testid="signature">Made by Andrew Moy</div>
            <div id="title" data-testid='title'>Battleship</div>
        </div>
    )
}