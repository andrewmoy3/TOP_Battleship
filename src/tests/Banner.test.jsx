/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react'
import Banner from '../Components/Banner'

describe('renders the banner properly', () => {
    it('snapshot', () => {
        const { container } = render(<Banner />)
        expect(container).toMatchSnapshot()
    })

    it("contains the title 'Battleship'", () => {
        render(<Banner />)
        const titleElement = screen.getByText('battleship', { exact: false })
        expect(titleElement).toBeInTheDocument()
        expect(titleElement.textContent).toMatch('Battleship')
    })

    it('contains a title', () => {
        render(<Banner />)
        const title = screen.getByTestId('title')
        expect(title).toBeInTheDocument()
    })

    it('contains a signature', () => {
        render(<Banner />)
        const signature = screen.getByTestId('signature')
        expect(signature).toBeInTheDocument()
    })
})
