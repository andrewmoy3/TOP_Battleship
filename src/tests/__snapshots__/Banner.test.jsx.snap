// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders the banner properly > snapshot 1`] = `
<div>
  <div
    class="banner"
    id="banner"
  >
    <div
      data-testid="signature"
      id="signature"
    >
      Made by Andrew Moy
    </div>
    <div
      data-testid="title"
      id="title"
    >
      Battleship
    </div>
  </div>
</div>
`;
